subsys/net/lib/fota_download/src/fota_download.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdint.h>
+#include <zephyr.h>
+#include <flash.h>
+#include <download_client.h>
+#include <dfu_data_writer.h>
+#include <dfu/mcuboot.h>
+#include <pm_config.h>
+#include <logging/log.h>
+#include <fota_download.h>
+
+LOG_MODULE_REGISTER(fota_download, CONFIG_FOTA_DOWNLOAD_LOG_LEVEL);
+
+static		fota_download_callback_t callback;
+static struct	download_client dfu;
+
+static int download_client_callback(const struct download_client_evt *event)
+{
+	int err;
+
+	__ASSERT(event != NULL, "invalid dfu object\n");
+
+	switch (event->id) {
+	case DOWNLOAD_CLIENT_EVT_FRAGMENT: {
+		size_t size;
+
+		err = dfu_data_writer_verify_size(size);
+		if (err) {
+			LOG_ERR("Requested file too big to fit in flash\n");
```

Comment:
All across this file: no need for `\n` in log messages - logger will do it for you.